composer

```diff
@@ -11,6 +11,149 @@
  * @version 1.1.0
  */
 if (!class_exists('ComposerWrapper')) {
+    class ComposerWrapperParams {
+        /**
+         * @var int|false
+         */
+        private $forceMajorVersion = false;
+        /**
+         * @var string
+         */
+        private $updateFreq = '7 days';
+        /**
+         * @var string
+         */
+        private $composerDir = __DIR__;
+        /**
+         * @var string
+         */
+        private $wrapperDir = __DIR__;
+
+        public function __construct($wrapperDir = null)
+        {
+            if (null !== $wrapperDir) {
+                $this->wrapperDir = $wrapperDir;
+            }
+        }
+
+        public function loadReal()
+        {
+            $this->loadComposerJson();
+            // env variables have the highest priority
+            $this->loadEnv();
+        }
+
+
+        private function loadEnv()
+        {
+            if ( false !== $value = \getenv('COMPOSER_UPDATE_FREQ')) {
+                $this->setUpdateFreq($value);
+            }
+            if ( false !== $value = \getenv('COMPOSER_FORCE_MAJOR_VERSION')) {
+                $this->setForceMajorVersion($value);
+            }
+            if ( false !== $value = \getenv('COMPOSER_DIR')) {
+                $this->setComposerDir($value);
+            }
+        }
+
+        private function loadComposerJson()
+        {
+            $file = $this->wrapperDir . '/composer.json';
+            // It is the same logic of reading composer.json as is in the Composer
+            if (!is_file($file) || !is_readable($file) || !is_array($composer = json_decode(file_get_contents($file),                                                                                        true))) {
```

Comment:
Until I scrolled it all the way to the right, I was wondering how this code even works :roll_eyes: 

![image](https://user-images.githubusercontent.com/231518/98161235-75aa2000-1ef0-11eb-84ac-8869d6ca49da.png)

Reply:
oops